### PR TITLE
Implement json converter

### PR DESCRIPTION
**Pull request by kamilsa**
Signed-off-by: kamilsa <kamilsa16@gmail.com>

### Description of the Change
Adds json converters from shared model objects with proto backend. 

### Benefits

Allows converting shared model objects into json and vice versa by means of protobuf methods.

### Possible Drawbacks 

* Currently conversions are done using functions without classes. Might look as a drawback for someone.

* While model to json function is straightforward, conversion from command contains 3 functions: 
1. jsonToModel -- general function to convert json into protobuf message objects. Returns none if json is not valid.
2. jsonToBlock -- function converting json into shared model block. Returns none if json is not valid.
3. jsonToTransaction -- function converting json into shared model transaction. Returns none if json is not valid.

* At the moment this PR was implemented proto class for proposal was not implemented. In future jsonToProposal function might be included.

### Usage Examples or Tests *[optional]*

Tests were added.

### CODE REVIEW

File: shared_model/converters/protobuf/json_proto_converter.hpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    namespace protobuf {
+
+      /**
+       * Converts protobuf model object into json string
+       * @tparam T is the type of converting command
+       * @param command is the command to be converted
+       * @return json string
+       */
+      template <typename T>
+      std::string modelToJson(T command) {
```
Comment (x3medima17) [inline shared_model/converters/protobuf/json_proto_converter.hpp]: Does it convert any model object to JSON, or only `command`?

File: shared_model/converters/protobuf/json_proto_converter.hpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+       * @param json is the json string containing transaction
+       * @return optional of shared model transaction object, containing the
+       * object if conversion was successful and none otherwise
+       */
+      boost::optional<shared_model::proto::Transaction> jsonToTransaction(
+          std::string json) {
+        auto tx = jsonToModel<iroha::protocol::Transaction>(json);
+        if (tx) {
+          return shared_model::proto::Transaction(
+              std::move(tx.value()));
```
Comment (muratovv) [inline shared_model/converters/protobuf/json_proto_converter.hpp]: Is it possible to fetch failure result from the tx object?

File: shared_model/converters/protobuf/json_proto_converter.hpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        }
+        return boost::none;
+      }
+      /**
+       * Converts json into block shared model object
+       * @param json is the json string containing block
+       * @return optional of shared model transaction object containing the
+       * object if conversion was successful and none otherwise
+       */
+      boost::optional<shared_model::proto::Block> jsonToBlock(
```
Comment (muratovv) [inline shared_model/converters/protobuf/json_proto_converter.hpp]: This and previous method looks so similar. Maybe better to make a template method for them both?

File: test/module/shared_model/converters/json_proto_converter_test.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+#include "backend/protobuf/transaction.hpp"
+#include "converters/protobuf/json_proto_converter.hpp"
+
+using namespace shared_model::converters::protobuf;
+
+/**
+ * creates sample transaction
+ * @return tx shared model object
+ */
+shared_model::proto::Transaction generateTransaction() {
```
Comment (muratovv) [inline test/module/shared_model/converters/json_proto_converter_test.cpp]: You should remove this method and use builders instead.

File: test/module/shared_model/converters/json_proto_converter_test.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+  return shared_model::proto::Transaction(
+      std::move(proto_tx));
+}
+
+/**
+ * creates sample block containing several txs
+ * @return block shared model object
+ */
+shared_model::proto::Block generateBlock() {
```
Comment (muratovv) [inline test/module/shared_model/converters/json_proto_converter_test.cpp]: Same as before.

File: test/module/shared_model/converters/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#
+# Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
```
Comment (l4l) [inline test/module/shared_model/converters/CMakeLists.txt]: wrong year

**Pull request merged**